x/group/module/module.go

```diff
@@ -0,0 +1,180 @@
+package module
+
+import (
+	"encoding/json"
+	"math/rand"
+
+	"github.com/gorilla/mux"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/spf13/cobra"
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	sdkclient "github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	"github.com/cosmos/cosmos-sdk/x/group"
+	groupkeeper "github.com/cosmos/cosmos-sdk/x/group/keeper"
+)
+
+var (
+	_ module.AppModule           = AppModule{}
+	_ module.AppModuleBasic      = AppModuleBasic{}
+	_ module.AppModuleSimulation = AppModule{}
+)
+
+type AppModule struct {
+	AppModuleBasic
+	keeper        groupkeeper.Keeper
+	BankKeeper    group.BankKeeper
+	AccountKeeper group.AccountKeeper
+	registry      cdctypes.InterfaceRegistry
+}
+
+// NewAppModule creates a new AppModule object
+func NewAppModule(cdc codec.Codec, keeper groupkeeper.Keeper, ak group.AccountKeeper, bk group.BankKeeper, registry cdctypes.InterfaceRegistry) AppModule {
+	return AppModule{
+		AppModuleBasic: AppModuleBasic{cdc: cdc},
+		keeper:         keeper,
+		BankKeeper:     bk,
+		AccountKeeper:  ak,
+		registry:       registry,
+	}
+}
+
+type AppModuleBasic struct {
+	cdc codec.Codec
+}
+
+// Name returns the group module's name.
+func (AppModuleBasic) Name() string {
+	return group.ModuleName
+}
+
+// DefaultGenesis returns default genesis state as raw bytes for the group
+// module.
+func (AppModuleBasic) DefaultGenesis(cdc codec.JSONCodec) json.RawMessage {
+	// TODO: return default genesis state
+	return nil
+}
+
+// ValidateGenesis performs genesis state validation for the group module.
+func (AppModuleBasic) ValidateGenesis(cdc codec.JSONCodec, config sdkclient.TxEncodingConfig, bz json.RawMessage) error {
+	// TODO: perform genesis validation
+	return nil
+}
+
+// GetQueryCmd returns the cli query commands for the group module
+func (AppModuleBasic) GetQueryCmd() *cobra.Command {
+	// TODO: return CLI query commands
+	return nil
+}
+
+// GetTxCmd returns the transaction commands for the group module
+func (AppModuleBasic) GetTxCmd() *cobra.Command {
+	// TODO: return CLI tx commands
+	return nil
+}
+
+// RegisterGRPCGatewayRoutes registers the gRPC Gateway routes for the group module.
+func (a AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx sdkclient.Context, mux *runtime.ServeMux) {
+}
```

Comment:
BTW, why don't we have grpc-gateway routes? Or will they be added later?

Reply:
yeah I was planning to add an issue for adding that, will do that now

Reply:
#10661